pkg/apis/core/types.go

```diff
@@ -425,25 +425,51 @@ type PersistentVolumeClaimList struct {
 // PersistentVolumeClaimSpec describes the common attributes of storage devices
 // and allows a Source for provider-specific attributes
 type PersistentVolumeClaimSpec struct {
-	// Contains the types of access modes required
+	// Contains the types of access modes the volume should have. Valid access
+	// modes include:
+	// * ReadWriteOnce - can be mounted read/write mode to exactly 1 host.
+	// * ReadOnlyMany - can be mounted in read-only mode to many hosts.
+	// * ReadWriteMany - can be mounted in read/write mode to many hosts.
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1
 	// +optional
 	AccessModes []PersistentVolumeAccessMode
 	// A label query over volumes to consider for binding. This selector is
-	// ignored when VolumeName is set
+	// ignored when volumeName is set.
+	// A claim with a non-empty selector can't have a PersistentVolume
+	// dynamically provisioned for it.
 	// +optional
 	Selector *metav1.LabelSelector
-	// Resources represents the minimum resources required
+	// Represents the minimum resources the volume should have.
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#resources
 	// +optional
 	Resources ResourceRequirements
-	// VolumeName is the binding reference to the PersistentVolume backing this
-	// claim. When set to non-empty value Selector is not evaluated
+	// The binding reference to the PersistentVolume backing this claim. When
+	// set to non-empty value, selector is not evaluated.
 	// +optional
 	VolumeName string
-	// Name of the StorageClass required by the claim.
-	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1
+	// Name of the StorageClass required by the claim. Only PersistentVolumes of
+	// the requested class can be bound to the claim.
+	// If the value is "" (an empty string), the claim can only be bound to a
+	// PersistentVolume with no StorageClass (the storageClassName field is
+	// either unset or set to "").
+	// If this field is not set, the behavior depends on whether the
+	// DefaultStorageClass admission plugin is turned on.
+	// * If the admission plugin is turned on, the administrator may specify a
+	//   default StorageClass. This claim can be bound only to PersistentVolumes
```

Comment:
What happens if no default storageclass exists?

Reply:
Oops over-deleted things. I'm also planning to add back the behavior when multiple storageclasses exist